src/ast_editor.ml

```diff
@@ -0,0 +1,549 @@
+open Import
+open Ppx_utils
+
+let webview_map = ref (Map.empty (module String))
+
+let original_mode = ref true
+
+let hover_disposable_ref = ref None
+
+let doc_string_uri ~document = Uri.toString (TextDocument.uri document) ()
+
+let pp_doc_to_changed_origin_map = ref (Map.empty (module String))
+
+let origin_to_pp_doc_map = ref (Map.empty (module String))
```

Comment:
All of these global variables should be moved to a record. The record should then be attached to the global state of the plugin. This will guarantee that this is all cleaned up correctly once the plugin is unloaded.

Reply:
Did i do it right? 692b39a1d0a5c5fa7bad7c99864a1cf187309f71

Reply:
Yes, it looks much better.